lib/puppet/util/pidlock.rb

```diff
@@ -85,7 +85,7 @@ def clear_if_stale
       end
 
       args = Puppet::Util::Execution.execute(["ps", ps_argument, lock_pid, "-o", "args="]).strip
-      @lockfile.unlock unless procname =~ /ruby/ && args =~ /puppet/ || procname =~ /puppet(-.*)?$/
+      @lockfile.unlock unless procname =~ /ruby/ && (args =~ /puppet/ || args =~ /util\/rspec_runner/) || procname =~ /puppet(-.*)?$/
```

Comment:
@GabrielNagy wanted to get your thoughts about this. One thing that seems odd to me is if `Process.kill` succeeds, then we know there is another process with pid `lockpid` currently running. So why not check if `lock_pid == Process.pid` before doing the above? If the lock_pid is us, then we can be pretty sure the lock isn't stale, and we return before getting to line 76. Thoughts?

Reply:
That seems logical. Didn't know this could usually happen but from what I see it's almost always the case with the ssl lockfile.

Reply:
I pulled the "stale pid" commit into a separate PR and targeted that for 5.5.x https://github.com/puppetlabs/puppet/pull/8086. I didn't add a check for `lock_pid == Process.pid`, but if that's something you want to add, please do.